File: Languages/Java/Day017.java
// Languages - Java - Day 17
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that generates the Fibonacci sequence up to a given number n. 
// The Fibonacci sequence starts with 0 and 1, and each subsequent number is the sum of the previous two numbers.

import java.util.Scanner;

public class Day017 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a positive number: ");

        // Input Validation
        if (sc.hasNextInt()) {
            int n = sc.nextInt();
            if (n >= 0) {
                System.out.println("Fibonacci sequence up to " + n + ":");
                fibonacciUpToN(n);
            } else {
                System.out.println("Please enter a positive integer.");
            }
        } else {
            System.out.println("Invalid input. Please enter a positive integer.");
        }
        sc.close();
    }

    // Generates Fibonacci sequence up to the given number n
    public static void fibonacciUpToN(int n) {
        int a = 0, b = 1;

        if (n == 0) {
            System.out.print("0");
            return;
        }

        System.out.print(a + " ");
        while (b <= n) {
            System.out.print(b + " ");
            int sum = a + b;
            a = b;
            b = sum;
        }
        System.out.println();
    }
}

File: Languages/C++/Day017.cpp
// Languages - C++ - Day 17
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that generates the Fibonacci sequence up to a given number n.
// The Fibonacci sequence starts with 0 and 1, and each subsequent number is the sum of the previous two numbers.

#include <iostream>
#include <limits>

void fibonacci(int n)
{
    if (n < 0)
    {
        std::cout << "Please enter a positive integer.\n";
        return;
    }
    else if (n == 0)
    {
        std::cout << "Fibonacci sequence up to 0: 0\n";
        return;
    }

    int a = 0, b = 1, c;
    std::cout << "Fibonacci sequence up to " << n << ": " << a;

    if (n >= 1)
    {
        std::cout << ", " << b;
    }

    c = a + b;
    while (c <= n)
    {
        std::cout << ", " << c;
        a = b;
        b = c;
        c = a + b;
    }
    std::cout << std::endl;
}

int main()
{
    int n;
    std::cout << "Enter a positive number: ";

    // Input Validation
    if (!(std::cin >> n))
    {
        std::cout << "Invalid input. Please enter a positive integer.\n";
        return 1;
    }

    fibonacci(n);
    return 0;
}
